hydrolib/tools/extforce_convert/converters.py

```diff
@@ -759,6 +804,115 @@ def contains(quantity_class, quantity) -> bool:
+    """
+
+    @staticmethod
+    def convert(cmp_model: CMPModel) -> List[ForcingBase]:
+        """
+        Convert a CmpModel into a ForcingModel.
+
+        Args:
+            cmp_model (CmpModel):
+                The input CmpModel to be converted.
+
+        Returns:
+            ForcingModel: The converted ForcingModel.
+
+        Raises:
+            ValueError: If the lengths of the columns in the first row of the CmpModel do not match the number of
+                quantities in the CmpModel.
+
+        Examples:
+            Convert a CmpModel into a ForcingModel.
+                ```python
+                >>> data = {
+                ...     "comments": ["# Example comment"],
+                ...     "components": [
+                ...         {
+                ...             "harmonics": [{"period": 0.0, "amplitude": 1.0, "phase": 2.0}],
+                ...         }, {
+                ...             "astronomics": [{"name": "4MS10", "amplitude": 1.0, "phase": 2.0}],
+                ...         }
+                ...     ],
+                ...     "quantities_names": ["discharge", "waterlevel"],
+                ... }
+                >>> cmp_model = CMPModel(**data)
+                >>> forcing_model = CMPToForcingConverter.convert(cmp_model)
+                >>> forcing_model[0].datablock
+                [[0.0, 1.0, 2.0]]
+                >>> forcing_model[1].datablock
+                [['4MS10', 1.0, 2.0]]
+
+                ```
+        """
+        forcing_list = []
+        units = cmp_model.get_units()
+
+        for i in range(len(cmp_model.components)):
+            harmonic_model = CMPToForcingConverter.convert_harmonic(
+                cmp_model.quantities_names[i],
+                cmp_model.components[i].harmonics,
+                units[i],
+            )
+            if harmonic_model:
+                forcing_list.append(harmonic_model)
+
+            astronomic_model = CMPToForcingConverter.convert_astronomic(
+                cmp_model.quantities_names[i],
+                cmp_model.components[i].astronomics,
+                units[i],
+            )
+            if astronomic_model:
+                forcing_list.append(astronomic_model)
+
+        return forcing_list
+
+    @staticmethod
+    def convert_harmonic(
+        quantity_name: str, harmonics: List[HarmonicRecord], unit: str
+    ) -> Optional[Harmonic]:
+        harmonic_block = [
+            [harmonic.period, harmonic.amplitude, harmonic.phase]
+            for harmonic in harmonics
+        ]
+        if harmonic_block:
+            harmonic_model = Harmonic(
+                name=quantity_name,
+                function="harmonic",
+                quantityunitpair=[
+                    QuantityUnitPair(quantity="harmonic component", unit="minutes"),
+                    QuantityUnitPair(quantity=quantity_name, unit=unit),
+                    QuantityUnitPair(quantity=quantity_name, unit="deg"),
```

Comment:
I suspect it must be corrected to:
```suggestion
                    QuantityUnitPair(quantity=quantity_name + " amplitude", unit=unit),
                    QuantityUnitPair(quantity=quantity_name + " phase", unit="deg"),
```

Reply:
applied